homeassistant/components/hue/light.py

```diff
@@ -406,10 +406,27 @@ def device_info(self):
         else:
             command['alert'] = 'none'
 
-        if self.is_group:
-            await self.light.set_action(**command)
-        else:
-            await self.light.set_state(**command)
+        await self.async_set_command(**command)
+
+    async def async_set_command(self, **command):
+        """Send command to light or group and retry on error."""
+        from aiohue.errors import RequestError, ResponseError
+
+        for i in range(0, 3):
+
+            try:
+                sleep(i * 0.5)
+                if self.is_group:
+                    await self.light.set_action(**command)
+                else:
+                    await self.light.set_state(**command)
+                break
+
+            except (RequestError, ResponseError) as ex:
```

Comment:
When would response error be raised?

Reply:
I had this occuring when I tested it… sometimes the bridge returned an invalid response (text/html) when it was hammered. Didn't saw the content though, as I didn't run HA in debug mode.

Reply:
If it is of any assistance: received that error yesterday also (text/html) , had not seen it before though.

Error displayed without further info or text for that matter. 

Ha 90.2 here.

Reply:
@balloob I know catching those errors is a bit… generous. But I couldn't find a fast way to differ between an API rate limit error (which results in a `RequestError`) and any other request errors (which also result in a `RequestError`). The same for `ResponseError`. If the error persists over 3 tries, then it will be raised anyway.

What are your opinions here? I just had a deeper look at your  [aiohue](https://github.com/balloob/aiohue) project and here are my thoughts:

It would be much nicer if we identify the specific API rate limit error (probably by the HTTP return code & content) and raise a specific `Exception` in the `aiohue` module. That way, would could catch it in HASS and to exception handling only for this case. 

I also had a look at the exception handling in `aiohue`. It seems that `aiohttp` is raising exceptions with a lot of context. Unfortunately, `aiohue` is wrapping all those exceptions and throwing away all the context (merging all exceptions back into one, removing all informations, not using `raise… from…` syntax properly).

I opened a new [issue in aiohue](https://github.com/balloob/aiohue/issues/17) to address this concern. It will probably help in exactly this case, but if you've a better idea how to handle it, I'm open for it :)